### PR TITLE
Initial version of xt-distro

### PR DESCRIPTION
**Pull request by andr2000**

### CODE REVIEW

File: meta/conf/machine/h3ulcb.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+require xt-distro-machine.inc
+
+MACHINEOVERRIDES =. "rcar:rcar-gen3:"
```
Comment (lorc) [inline meta/conf/machine/h3ulcb.conf]: how do you think, should we also provide chip type (e.g. H3, M3)?

Comment (andr2000) [inline meta/conf/machine/h3ulcb.conf]: yes, someday we will need to do so, but at the moment it's fine: once we start developing real products we most probably will update this to fit actual requirements

**Comment by lorc** [comment]
Okay, I'm fine with this.

`Reviewed-by: Volodymyr Babchuk <vlad.babchuk@gmail.com>`

**Comment by andr2000** [comment]
poky.git/meta/classes$ cat base.bbclass | wc -l
702
xt-distro/meta/classes$ cat base.bbclass | wc -l
176
and intersection is rather small

File: meta/classes/build_yocto.bbclass [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# N.B. this is guaranteed by the layer priority (see layer.conf
+# BBFILE_PRIORITY_xt-core = "5") that this append will
+# run after base_do_unpack and BEFORE any other unpacks
+# such as proprietary code unpack in other layers etc.
+# so, the repo will get updated properly before other
+# related recipes
+build_yocto_repo_sync() {
```
Comment (aanisov) [inline meta/classes/build_yocto.bbclass]: I guess this should be moved out build_yocto.bbclass to some generic override for repo fetcher (also with separate commit). You need the same functionality for any stuff you will be getting using .repo, i.e. android sources.

Comment (andr2000) [inline meta/classes/build_yocto.bbclass]: bitbake fetcher for repo:// is a part of BB python library, so you cannot override it
and you have to modify BB library which I don't want.
as to Android build, we will probably move some common code for Yocto and Android builds into a common class, but at the moment it seems to be kind of premature optimization

Comment (aanisov) [inline meta/classes/build_yocto.bbclass]: IMHO it is already known piece of common code worth to be separated.

**Pull request merged**